kubespawner/reflector.py

```diff
@@ -203,12 +205,27 @@ def __init__(self, *args, **kwargs):
         if not self.list_method_name:
             raise RuntimeError("Reflector list_method_name must be set!")
 
-        self.start()
+        self.watch_task = None  # Test this rather than absence of the attr
 
-    def __del__(self):
-        self.stop()
+        # Note that simply instantiating the class does not load
+        # Kubernetes config, acquire an API client, or start the
+        # watch task.
+
+    @classmethod
+    async def reflector(cls, *args, **kwargs):
+        """
+        This is how you should instantiate a reflector in order to bring
+        it up in a usable state, as this classmethod does load
+        Kubernetes config, acquires an API client, and starts the watch
+        task.
+        """
+        inst = cls(*args, **kwargs)
+        await load_config()
+        inst.api = shared_client(inst.api_group_name)
```

Comment:
Do we still need `shared_client`? The reason we had that for sync kubernetes is that each client object a ThreadPool, so the process exploded with threads if we kept creating clients (especially when it was an unconfigurable nthreads=n_cpus per client!). Do kubernetes_asyncio clients also have state we need to avoid creating too many times?

Reply:
I don't know if they have state, but we at least would be doing TLS connection establishment once per client, right?  I suspect at scale a shared client is a win, but I think we should keep the shared client for this PR anyway.